src/shared/utils.ts

```diff
@@ -911,27 +852,135 @@ export function getCommentIdFromProps(props: any): number | undefined {
   return id ? Number(id) : undefined;
 }
 
-export function editCommentRes(data: CommentView, comments?: CommentView[]) {
-  let found = comments?.find(c => c.comment.id == data.comment.id);
-  if (found) {
-    found.comment.content = data.comment.content;
-    found.comment.distinguished = data.comment.distinguished;
-    found.comment.updated = data.comment.updated;
-    found.comment.removed = data.comment.removed;
-    found.comment.deleted = data.comment.deleted;
-    found.counts.upvotes = data.counts.upvotes;
-    found.counts.downvotes = data.counts.downvotes;
-    found.counts.score = data.counts.score;
+function editListImmutable<T>(fieldName: string, data: T, list: T[]): T[] {
```

Comment:
I made a version of this with better typing:
```typescript
type FieldNames = "person_mention" | "community" | "private_message"; // etc.

function editListImmutable<T extends {[key in F]: {id: number}}, F extends FieldNames>(fieldName: F, data: T, list: T[]): T[] {
  const foundIndex = list.findIndex(c => c[fieldName].id === data[fieldName].id);
  if (foundIndex != -1) {
    const newList = list;
    newList[foundIndex] = data;
    return newList;
  } else {
    return list;
  }
}
```

Reply:
Thx, I'll update this shortly.